packages/angular_devkit/build_angular/src/webpack/plugins/scripts-webpack-plugin.ts

```diff
@@ -149,19 +154,36 @@ export class ScriptsWebpackPlugin {
         });
 
         const combinedSource = new webpackSources.CachedSource(concatSource);
-        const filename = interpolateName(
-          { resourcePath: 'scripts.js' },
-          this.options.filename as string,
-          {
-            content: combinedSource.source(),
-          },
-        );
-
-        const output = { filename, source: combinedSource };
+
+        const output = { filename: this.options.filename, source: combinedSource };
         this._insertOutput(compilation, output);
         this._cachedOutput = output;
         addDependencies(compilation, scripts);
       });
+      compilation.hooks.processAssets.tapPromise(
+        {
+          name: PLUGIN_NAME,
+          stage: compiler.webpack.Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE_HASH,
+          additionalAssets: true,
+        },
+        async (compilationAssets) => {
+          for (const assetName in compilationAssets) {
+            if (Object.prototype.hasOwnProperty.call(compilationAssets, assetName)) {
```

Comment:
We should only modify the asset that this plugin is handling.

```suggestion
           const assetName = this.options.filename;
```